src/components/ProductCreateEditPage/ProductCreateEditPage.tsx

```diff
@@ -0,0 +1,207 @@
+  isEditPage?: boolean;
+}
+
+export const ProductCreateEditPage = ({ isEditPage = false }: IProductCreateEditPageProps) => {
+  const { productId } = useParams();
+  const navigate = useNavigate();
+
+  // create page
+  const serviceContainerCreatePage = useServiceContainer(productApi.createProduct);
+
+  // edit page - get method
+  const serviceContainerEditPageGet = useServiceContainer(productApi.getProduct);
+  const serviceContainerEditPageGetRunner = serviceContainerEditPageGet.run;
+
+  // edit page - patch method
+  const serviceContainerEditPagePatch = useServiceContainer(productApi.patchProduct);
+
+  const { register, setFieldValues, getFieldState, getFieldErrors, handleSubmit, isSubmitDisabled } = useForm();
+
+  useTitle(
+    generatePageTitle({
+      pageType: isEditPage ? 'Edit' : 'Create',
+      serviceContainer: serviceContainerEditPageGet,
+      firstLevelEntity: 'Product',
+    })
+  );
+
+  const submitCreate = (data: IFieldValues) => {
+    return serviceContainerCreatePage
+      .run({
+        serviceData: { data },
+      })
+      .then((response: any) => {
+        const newProductId = response?.data?.id;
+        if (!newProductId) {
+          throw new Error(`Invalid productId coming from Orch POST response: ${newProductId}`);
+        }
+        navigate(`/products/${newProductId}`);
+      })
+      .catch((e: any) => {
+        throw new Error('Failed to create Product.');
+      });
+  };
+
+  const submitUpdate = (data: IFieldValues) => {
+    const patchData = createSafePatch(serviceContainerEditPageGet.data, data);
+
+    serviceContainerEditPagePatch
+      .run({ serviceData: { id: productId, patchData } })
+      .then(() => {
+        navigate(`/products/${productId}`);
+      })
+      .catch(() => {
+        throw new Error('Failed to edit Product.');
+      });
+  };
+
+  useEffect(() => {
+    if (isEditPage) {
+      serviceContainerEditPageGetRunner({ serviceData: { id: productId } }).then((response: any) => {
+        setFieldValues(response.data);
+      });
+    }
+  }, [isEditPage, productId, serviceContainerEditPageGetRunner, setFieldValues]);
+
+  const formComponent = (
+    <ContentBox padding isResponsive>
+      <Form
+        onSubmit={(e) => {
+          e.preventDefault();
+        }}
+      >
+        <FormGroup
+          isRequired
+          label={productEntityAttributes.name.title}
+          fieldId={productEntityAttributes.name.id}
+          helperText={
+            <FormHelperText isHidden={getFieldState(productEntityAttributes.name.id) !== 'error'} isError>
+              {getFieldErrors(productEntityAttributes.name.id)}
+            </FormHelperText>
+          }
+        >
+          <TextInput
+            isRequired
+            type="text"
+            id={productEntityAttributes.name.id}
+            name={productEntityAttributes.name.id}
+            autoComplete="off"
+            {...register<string>(productEntityAttributes.name.id, formConfig.name)}
```

Comment:
[optional]
```tsx
...register<string>(productEntityAttributes.name.id, formConfig.name)
```
could be changed to:
```tsx
... = useForm(formConfig);
// ...
...register<string>(productEntityAttributes.name.id)
```

It means `formConfig.name` would be automatically derived from `productEntityAttributes.name.id`, but then `formConfig` 1-level attribute keys would have to match entity attribute, TypeScript checker could be one of the possible solutions.

Reply:
Hi @patrikk0123 , can you mention here what was the final decision?